<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="keywords" content="hpc, research it, servers, clusters, storage" />
    <meta name="description" content="Lecture on HPC resources and methods." />
    <title>Orientation -- 2017-08-16</title>
    <style>
      @import url(https://fonts.googleapis.com/css?family=Droid+Serif);
      @import url(https://fonts.googleapis.com/css?family=Yanone+Kaffeesatz);
      @import url(https://fonts.googleapis.com/css?family=Ubuntu+Mono:400,700,400italic);

      body {
        font-family: 'Droid Serif';
      }
      h1, h2, h3 {
        font-family: 'Yanone Kaffeesatz';
        font-weight: 400;
        margin-bottom: 0;
      }
      .remark-slide-content h1 { font-size: 3em; }
      .remark-slide-content h2 { font-size: 2em; }
      .remark-slide-content h3 { font-size: 1.6em; }
      .footnote {
        position: absolute;
        bottom: 3em;
      }
      li p { line-height: 1.25em; }
      .red { color: #fa0000; }
      .large { font-size: 2em; }
      a, a > code {
        color: rgb(249, 38, 114);
        text-decoration: none;
      }
      code {
        background: #e7e8e2;
        border-radius: 5px;
      }
      .remark-code, .remark-inline-code { font-family: 'Ubuntu Mono'; }
      .remark-code-line-highlighted     { background-color: #373832; }
      .pull-left {
        float: left;
        width: 47%;
      }
      .pull-right {
        float: right;
        width: 47%;
      }
      .pull-right ~ p {
        clear: both;
      }
      #slideshow .slide .content code {
        font-size: 0.8em;
      }
      #slideshow .slide .content pre code {
        font-size: 0.9em;
        padding: 15px;
      }
      .inverse {
        background: #272822;
        color: #777872;
        text-shadow: 0 0 20px #333;
      }
      .inverse h1, .inverse h2 {
        color: #f3f3f3;
        line-height: 0.8em;
      }
      /* Two-column layout */
      .left-column {
        color: #777;
        width: 20%;
        height: 92%;
        float: left;
      }
        .left-column h2:last-of-type, .left-column h3:last-child {
          color: #000;
        }
      .right-column {
        width: 75%;
        float: right;
        padding-top: 1em;
      }
      .coolBorder {
        border: 2px solid purple;
        border-radius: 20px;
        padding: 20px;
      }
    </style>
  </head>
  <body>
    <textarea id="source">

class: center, middle

# HPC
<div style="text-align:center"><img src="images/captain-lionel-mandrake-high-performance-computing-or-how-i-learned-to-stop-worrying-and-love-linux.jpg" alt="High Performance Computer or: How I Learned to Stop Worrying and Love Linux"/></div>

???

---

# What / Who is ResearchIT at Iowa State?

* What do we do?
    * Collaborate with researchers to solve complex technology problems, freeing them up to focus on their research
    * Provide access to computational hardware, software, storage, consulting
    * Focus our time on tasks that scale

* Who?
    * Collaboration with LAS, CALS, ENGR, ITS
    * [http://researchit.las.iastate.edu](http://rit.las.iastate.edu)
    * Lots of involvement from other IT people in college, departments, and ITS
    * researchit@iastate.edu

---

# Pricing Model

* We do not charge for our time
    * Our salaries are paid through the colleges, departments, etc. that we support
      * Offloading liability
* We do not charge per CPU cycle
* We want to collaborate with your PI's on grant proposals to include funding for access to shared infrastructure
* We will do our best to find spare cycles to run your research if you don't have funding (may not be on the newest hardware)

---

# Data Storage

* Large Scale Storage (aka Research Files)
    * Stable, inexpensive storage for large research data
    * Located in Durham Data Center
    * Multiple nodes
    * File snapshots, daily for 2 months
    * Fast network access (40Gbps)
    * Providing each LAS PI 2TB of free storage
    * PI cost for additional storage is $40 / TB / Year (cheaper than Amazon Glacier)
    * For more info on LSS: [https://researchit.las.iastate.edu/large-scale-storage-lss](https://researchit.las.iastate.edu/large-scale-storage-lss)
* Please do not store your critical research data on your laptop or on external USB disk drives

---

# Computational Resources

* [Full List of hardware](https://rit.las.iastate.edu/hardware)
* RIT1 / RIT2:
    * Older machines, free for research use. All bcbgrads are on the list for rit1.
* BigRam:
    * for when your whole dataset needs to fit in RAM (e.g. de novo assembly)
* BioCrunch:
    * for multithreaded, non MPI software (mapping, alignment, etc.)
* Speedy / Speedy2:
    * Fewer faster cores
    * For single threaded software
* [Condo 2017 HPC Cluster](http://www.hpc.iastate.edu/guides/condo):
    * Approx 200 Nodes
    * Primarly for MPI capable software
    * For sponsored research (College / PI supported)
* [Condo 2017 Free Tier](http://www.hpc.iastate.edu/guides/condo):
    * 48 nodes + a few specialty nodes
    * This tier is to allow groups without funding to complete research, gain an understanding of the HPC environment, etc.
* Legion:
    * 2176 threads across 8 nodes
    * Meant for highly parallel workloads (molecular dynamics, genome alignment, Monte Carlo simulations)
* LAS-Win-01:
    * When software will only run on Windows
* [hpc-class](http://www.hpc.iastate.edu/guides/classroom-hpc-cluster)
    * For classes, not research

---
# Other Research Computing Resources

## [Xsede](https://www.xsede.org/)
* ISU researchers have access to the national supercompute centers (e.g. TACC, PSC) via Xsede
* For problems that need to scale larger than our on-campus resources
* Contact campus champoins: Jim Coyle or Andrew Severin

## Cloud (AWS, Azure, etc.)
* Tempting introductory rates
* Be cautious of putting in a credit card (charges can accumulate quickly)
* Consider data transfer times and speed
* Consult with IT before purchasing - we have special negotiated rates

---

# Older Compute Resources at ISU

* cyence
  * NSF MRI grant
* cystorm
  * Being retired
* lightning3
  * Being retired

---

# A note on operating systems:

* High Performance Computing resources you encounter will typically be running Linux
    * At ISU that will be Red Hat Enterprise Linux (RHEL)
* [Fedora](https://getfedora.org/) is the upstream desktop distribution (good option to become familiar with linux)
* [CentOS](https://www.centos.org/) is the downstream server distribution
* When you google linux questions, you'll inevitably get results for Ubuntu
    * results can point you in the right direction, but interpret and understand - don't just copy and paste
    * sudo apt-get install - will not work for you on our systems

---

# Software

## Modules
* We use modules to allow multiple people to use the same software with consistent, reproducible results
* Modules keep your environment clean and free of conflicts (e.g. python2/3 or Java7/8)
* Think of modules as a software library. You need to check out the software before you can use it.
* We call our software library [RISA](https://researchit.las.iastate.edu/research-it-software-archive) (Research IT Software Archive), and sync it to hpc-class, condo, and all of our custom machines
```
$ module avail
------------------------------------------------------------------ /opt/rit/modules ------------------------------------------------------------------
abyss/1.5.2              freesurfer/5.3.0         lib/htslib/1.2.1         microbiomeutil/r20110519 rsem/1.2.22
abyss/1.9.0              gapcloser/1.12-r6        lib/htslib/1.3           migrate-n/3.6.11         rsem/1.2.31
afni/17.0.10             gapfiller/1-10           lib/htslib/1.3.2         mira/4.0.2               rtax/0.984
albert/20170105          gatk/3.4-46              lib/ICE/1.0.9            mirdeep/2.0.0.7          ruby/2.3.0
...
```

## BioDatabases on Condo at : /work/LAS/BioDatabase
* BLASTdb, FASTA, iGenomes, mpiBLASTdb

---

# Software

## Modules

* To use a module
```
$ module load abyss
```
* Default behavior is to load the latest version if not specified
* Use 'module purge' to clear your environment before loading something different

---
# Consulting

* Do not charge for time, but we are also not your dedicated programmer or software installer
* Can advise on computational performance of software
* Can help you identify optimal software for your pipeline
* Help identify ways to optimize pipeline
    * Hardware
    * Different software
    * Parallelization wrappers (e.g. gnu parallel)
* Our goal is to work on tasks that can benefit others through packaged software, documentation, collaboration, etc. and that help you overcome technical challenges that moves your science forward
* Faster iterations

---

# Consulting Examples

* Hufford lab cut genome assembly time by running on BigRAM
* Helped Nikolau lab migrate from MEGA to RAxML
* Sivasankar lab cut MD (gromacs) job run time, and costs by moving off Amazon EC2 to Condo
* Helped Windus lab compile GAMESS on Condo
* Helped Dobbs lab publish web interface for rpimotif tool
* Helped Wolter lab parallelize matlab code
* Helped Lu lab port code from Windows to Linux

---

# Websites

* Lab Websites
* Custom web based research tools
* Secure virtual web environments
* Code review
* Our platform: http://www.biology-it.iastate.edu/luggage_doc/
* Need to publish a new service, or tool on the web ?

---

# Other Services

* Amazon Web Services:
* e.g. S3 buckets to host bigwig files for gbrowse, EC2 instances for compute jobs, GPU nodes
* Version Control:
* Can help you start tracking code in git, several options for hosting including on-site: http://it.las.iastate.edu/version-control
* Backups:
* We can set you up with backup software to protect data on your local disk: http://it.las.iastate.edu/computer-backup

---

# Tutorials


---

# Support

* [Research IT](http://rit.las.iastate.edu)
    * researchit@iastate.edu
    * IRC (chat): &#35;bitcom on freenode
* [ISU IT HPC Team](http://hpc.iastate.edu)
    * Condo & HPC-Class Support team
    * hpc-help@iastate.edu


    </textarea>
    <script src="https://gnab.github.io/remark/downloads/remark-latest.min.js" type="text/javascript">
    </script>
    <script>
      var hljs = remark.highlighter.engine;
    </script>
    <script src="remark.language.js"></script>
    <script>
      var slideshow = remark.create({
          highlightStyle: 'monokai',
          highlightLanguage: 'tex',
          highlightLines: true
        }) ;
    </script>
    <script>
      var _gaq = _gaq || [];
      _gaq.push(['_setAccount', 'UA-44561333-1']);
      _gaq.push(['_trackPageview']);

      (function() {
        var ga = document.createElement('script');
        ga.src = 'https://ssl.google-analytics.com/ga.js';
        var s = document.scripts[0];
        s.parentNode.insertBefore(ga, s);
      }());
    </script>
  </body>
</html>
